src/utils.ts

```diff
@@ -1840,27 +1870,52 @@ const MARKERS = ['+', 'crux', 'mark'];
 const CROSSES = ['x', 'cross', 'oblique'];
 const SQUARES = ['square'];
 const TRIANGLES = ['^', 'triangle', 'tri'];
+const STROKE_STYLE_OFFSET = 15;
 export class newPoint2D extends newShape {
   public path: Path2D;
   public center: Vertex;
 
   constructor(
     x: number = 0,
     y: number = 0,
-    private _size: number = 2,
-    private _marker: string = 'circle',
-    private _markerOrientation: string = 'up',
-    color?: string
+    protected _size: number = 2,
+    protected _marker: string = 'circle',
+    protected _markerOrientation: string = 'up',
+    fillStyle?: string,
+    strokeStyle?: string
   ) {
     super();
     this.center = new Vertex(x, y);
     this.path = this.buildPath();
-    if (color) this.strokeStyle = this.fillStyle = color;
+    this.fillStyle = fillStyle ? fillStyle : this.fillStyle;
+    this.strokeStyle = strokeStyle ? strokeStyle : this.setStrokeStyle(this.fillStyle);
+    this.lineWidth = 1;
   };
 
+  public getFillStyleHSL(fillStyle: string): [number, number, number] {
+    if (fillStyle.includes("rgb")) return arrayRGBToHSL(...RGBToArray(fillStyle));
+    return HSLToArray(fillStyle)
+  }
+
+  public getFillStyleRGB(fillStyle: string): [number, number, number] {
+    let [r, g, b] = fillStyle.split(',');
+    return [Number(r.split('rgb(')[1]), Number(g), Number(b.split(")")[0])]
+  }
+
+  public setStrokeStyle(fillStyle: string): string {
+    const [h, s, l] = this.getFillStyleHSL(fillStyle);
+    const lValue = l <= STROKE_STYLE_OFFSET ? l + STROKE_STYLE_OFFSET : l - STROKE_STYLE_OFFSET;
+    return `hsl(${h}, ${s}%, ${lValue}%)`;
+  }
+
+  public setColors(fillStyle?: string, strokeStyle?: string) {
+    this.fillStyle = fillStyle ? fillStyle : this.fillStyle;
```

Comment:
[Nullish coalescing operator](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing). Would require to set null as default value, which is better than the undefined you can have here

Reply:
Done